apps/server/src/modules/tool/controller/tool.controller.spec.ts

```diff
@@ -67,4 +113,191 @@ describe('ToolController', () => {
+				type: ToolConfigType.BASIC,
+				baseUrl: 'mockUrl',
+			});
+			const lti11ToolConfigResponse: Lti11ToolConfigResponse = new Lti11ToolConfigResponse({
+				key: 'mockKey',
+				lti_message_type: LtiMessageType.BASIC_LTI_LAUNCH_REQUEST,
+				privacy_permission: LtiPrivacyPermission.NAME,
+				type: ToolConfigType.LTI11,
+				baseUrl: 'mockUrl',
+			});
+			const oauth2ToolConfigResponse: Oauth2ToolConfigResponse = new Oauth2ToolConfigResponse({
+				clientId: 'mockId',
+				skipConsent: false,
+				type: ToolConfigType.OAUTH2,
+				baseUrl: 'mockUrl',
+			});
+			const externalToolResponse: ExternalToolResponse = new ExternalToolResponse({
+				id: '1',
+				name: 'mockName',
+				url: 'mockUrl',
+				logoUrl: 'mockLogoUrl',
+				parameters: [customParameterResponse],
+				isHidden: true,
+				openNewTab: true,
+				version: 1,
+				config: basicToolConfigResponse,
+			});
+
+			const basicToolConfigDO: BasicToolConfigDO = new BasicToolConfigDO({
+				type: ToolConfigType.BASIC,
+				baseUrl: 'mockUrl',
+			});
+			const lti11ToolConfigDO: Lti11ToolConfigDO = new Lti11ToolConfigDO({
+				secret: 'mockSecret',
+				key: 'mockKey',
+				lti_message_type: LtiMessageType.BASIC_LTI_LAUNCH_REQUEST,
+				privacy_permission: LtiPrivacyPermission.NAME,
+				type: ToolConfigType.LTI11,
+				baseUrl: 'mockUrl',
+			});
+			const oauth2ToolConfigDO: Oauth2ToolConfigDO = new Oauth2ToolConfigDO({
+				clientId: 'mockId',
+				skipConsent: false,
+				type: ToolConfigType.OAUTH2,
+				baseUrl: 'mockUrl',
+			});
+			const customParameterDO: CustomParameterDO = new CustomParameterDO({
+				name: 'mockName',
+				default: 'mockDefault',
+				location: CustomParameterLocation.PATH,
+				scope: CustomParameterScope.SCHOOL,
+				type: CustomParameterType.STRING,
+				regex: 'mockRegex',
+			});
+			const externalToolDO: ExternalToolDO = new ExternalToolDO({
+				id: '1',
+				name: 'mockName',
+				url: 'mockUrl',
+				logoUrl: 'mockLogoUrl',
+				parameters: [customParameterDO],
+				isHidden: true,
+				openNewTab: true,
+				version: 1,
+				config: basicToolConfigDO,
+			});
+
+			return {
+				body,
+				bodyConfigCreateBasicParams,
+				bodyConfigCreateOauthParams,
+				bodyConfigCreateLti11Params,
+				currentUser,
+				externalToolResponse,
+				externalToolDO,
+				lti11ToolConfigDO,
+				lti11ToolConfigResponse,
+				oauth2ToolConfigResponse,
+				oauth2ToolConfigDO,
+			};
+		}
+
+		it('should return basic external tool response', async () => {
+			const { body, bodyConfigCreateBasicParams, currentUser, externalToolResponse, externalToolDO } = setup();
+			body.config = bodyConfigCreateBasicParams;
+			externalToolMapper.mapRequestToExternalToolDO.mockReturnValue(externalToolDO);
+			externalToolUc.createExternalTool.mockResolvedValue(externalToolDO);
+			externalToolResponseMapper.mapToResponse.mockReturnValue(externalToolResponse);
+			const expected = await controller.createExternalTool(body, currentUser);
```

Comment:
Maybe you could organize you tests for more readability with extra lines between setup and preparation, action and expection?

Reply:
done